common/rfb/pam.c

```diff
@@ -84,8 +87,22 @@ int do_pam_auth(const char *service, const char *username, const char *password)
   };
   pam_handle_t *h = 0;
   ret = pam_start(service, username, &conv, &h);
-  if (ret == PAM_SUCCESS)
+  if (ret == PAM_SUCCESS) {
+
+#ifdef PAM_XDISPLAY
+    /* Pass the display name to PAM modules. */
+    ret = pam_set_item(h, PAM_XDISPLAY, displayStr);
+    /* Note: PAM_XDISPLAY is only supported by modern versions of PAM */
+    if (ret== PAM_SUCCESS || ret == PAM_BAD_ITEM) {
+        ret = pam_authenticate(h, 0);
```

Comment:
This `if`-structure will let users pass on without authentication if something unexpected happens with `pam_set_item()`.

Let's restructure this so that we don't duplicate the call to `pam_authenticate()`.